src/kvstore/RocksEngine.cpp

```diff
@@ -204,6 +205,7 @@ ResultCode RocksEngine::range(const std::string& start,
 ResultCode RocksEngine::prefix(const std::string& prefix,
                                std::unique_ptr<KVIterator>* storageIter) {
     rocksdb::ReadOptions options;
+    options.prefix_same_as_start = true;
```

Comment:
According to the explain of `prefix_same_as_start` 

```
This option is effective only for prefix seeks, i.e. prefix_extractor is non-null for the column family and total_order_seek is false.
```

Reply:
Hi, you mean we should first judge whether prefix is null？rocksdb already judge this.
Code in db/db_iter.cc like
```
DBIter::DBIter(Env* _env, const ReadOptions& read_options,
      ...
      prefix_same_as_start_(mutable_cf_options.prefix_extractor
                                ? read_options.prefix_same_as_start
                                : false),
```

Reply:
When you setting `prefix_same_as_start` as `true`, maybe you should make sure `prefix_extractor` is not null and `total_order_seek` is false ?

Reply:
Hi, **total_order_seek** is false by default. And as the code above, if prefix_extractor is null, **prefix_same_as_start_** will be false, no matter what **prefix_same_as_start** is.